R/splitBy.R

```diff
@@ -0,0 +1,181 @@
+#' Split \code{SummarizedExperiment} by grouping variables
+#'
+#' @param x A
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object
+#'
+#' @param grouping A single character value for selecting the grouping variable
+#'   from \code{colData} or \code{rowData}.
+#' 
+#' @param direction A single character value for selecting from where grouping
+#'   variable should be searched. Must be "col" or "row". 
+#'   
+#' @param ... 
+#' \itemize{
+#'   \item{\code{use_names} A boolean value to select whether to name elements of
+#'   list by their group names.}
+#' }
+#'
+#' @return
+#' List of \code{SummarizedExperiment} objects.
+#'
+#'
+#' @details
+#' \code{splitBy} split data based on grouping variable. Splitting can be done
+#' column-wise or row-wise. Returned value is a list of \code{SummarizedExperiment}
+#' objects; each element containing members of each group.
+#'
+#' @seealso
+#' \code{\link[mia::splitByRanks]{splitByRanks}},
+#' \code{\link[mia::unSplitByRanks]{unSplitByRanks}},
+#' \code{\link[=merge-methods]{mergeRows}},
+#' \code{\link[scuttle:sumCountsAcrossFeatures]{sumCountsAcrossFeatures}},
+#' \code{\link[=agglomerate-methods]{agglomerateByRank}},
+#' \code{\link[SingleCellExperiment:altExps]{altExps}},
+#' \code{\link[SingleCellExperiment:splitAltExps]{splitAltExps}}
+#'
+#' @name splitBy
+#' @export
+#' 
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' data(GlobalPatterns)
+#' tse <- GlobalPatterns
+#' # Split data based on SampleType. 
+#' se_list <- splitBy(tse, grouping = "SampleType")
+#' 
+#' se_list[1:2]
+#' 
+#' # Create arbitrary groups
+#' colData(tse)$group <- sample(1:10, ncol(tse), replace = TRUE)
+#' rowData(tse)$group <- sample(1:10, nrow(tse), replace = TRUE)
```

Comment:
Splitting by Phylum could be one possible use case. Can we add one example on that?